src/reference-manual/compiler-optimizations.Rmd

```diff
@@ -0,0 +1,487 @@
+The Stanc3 compiler can attempt to optimize a Stan program as it is compiled. The optimized program has the same behavior as the unoptimized program, but it may be faster, more memory efficient, or more numerically stable.
```

Comment:
I think this whole page now makes more sense as a sub-section of src/stan-users-guide/using-stanc.Rmd, which didn't exist when the PR was first opened